MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/CreateRelativePermeabilityModel.cpp

```diff
@@ -0,0 +1,237 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   CreateRelativePermeabilityModel.cpp
+ *
+ * Created on November 2, 2016, 11:43 AM
+ */
+
+#include "CreateRelativePermeabilityModel.h"
+
+#include <array>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+
+#include "MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h"
+
+#include "RelativePermeability.h"
+#include "RelativePermeabilityCurve.h"
+#include "WettingPhaseVanGenuchten.h"
+#include "NonWettingPhaseVanGenuchten.h"
+#include "WettingPhaseBrookCoreyOilGas.h"
+#include "NonWettingPhaseBrookCoreyOilGas.h"
+
+namespace MaterialLib
+{
+namespace PorousMedium
+{
+/**
+    \param config ConfigTree object which contains the input data
+                  including <type>WettingPhaseVanGenuchten</type>
+                  and it has a tag of <relative_permeability>
+*/
+std::unique_ptr<RelativePermeability> createWettingPhaseVanGenuchten(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material__porous_medium__relative_permeability__WettingPhaseVanGenuchten}
+    config.checkConfigParameter("type", "WettingPhaseVanGenuchten");
+
+    //! \ogs_file_param{material__porous_medium__relative_permeability__WettingPhaseVanGenuchten__sr}
+    const double Sr = config.getConfigParameter<double>("sr");
+
+    //! \ogs_file_param{material__porous_medium__relative_permeability__WettingPhaseVanGenuchten__smax}
+    const double Smax = config.getConfigParameter<double>("smax");
+
+    //! \ogs_file_param{material__porous_medium__relative_permeability__WettingPhaseVanGenuchten__m}
+    const double m = config.getConfigParameter<double>("m");
+    if (m < 0. || m > 1.0)  // m <= 1
```

Comment:
the comment does not match the expression. I'd remove the comments; the reason is already given in the error message.